_posts/2018-05-25-ecs-calculator.md

```diff
@@ -0,0 +1,108 @@
+  |----------------------|
+  |r3.large| 2048| 15299|
+  |r3.xlarge| 4096| 30680|
+  |r3.2xlarge| 8192| 61442|
+  |r3.4xlarge| 16384| 122950|
+  |r3.8xlarge| 32768| 245996|
+  |----------------------|
+  |i2.xlarge| 4096| 30680|
+  |i2.2xlarge| 8192| 61442|
+  |i2.4xlarge| 16384| 122950|
+  |i2.8xlarge| 32768| 245996|
+  |----------------------|
+  |i3.large| 2048| 15292|
+  |i3.xlarge| 4096| 30664|
+  |i3.2xlarge| 8192| 61408|
+  |i3.4xlarge| 16384| 122881|
+  |i3.8xlarge| 32768| 245857|
+  |i3.16xlarge| 65536| 491809|
+  |----------------------|
+  |m5.large| 2048| 7690|
+  |m5.xlarge| 4096| 15586|
+  |m5.2xlarge| 8192| 31377|
+  |m5.4xlarge| 16384| 62960|
+  |m5.12xlarge| 49152| 189294|
+  |m5.24xlarge| 98304| 378665|
+  |----------------------|
+  |c5.large| 2048| 3714|
+  |c5.xlarge| 4096| 7634|
+  |c5.2xlarge| 8192| 15473|
+  |c5.4xlarge| 16384| 31152|
+  |c5.9xlarge| 36864| 70351|
+  |c5.18xlarge| 73728| 140780|
+  |----------------------|
+  |d2.xlarge| 4096| 30664|
+  |d2.2xlarge| 8192| 61408|
+  |d2.4xlarge| 16384| 122881|
+  |d2.8xlarge| 36864| 245857|
+
+  </div>
+
+You can also find this information in yaml format [here](/assets/files/2018-05-25-ecs-calculator/ec2-instance-list.yml)
+
+## How the data can help you to optimally scale and fit ECS containers in an EC2 instance?
+
+
+The number of CPU units specifies the minimum amount of CPU reserved for a container. The CPU units are effectively allocated in containers on an EC2 instance. One container can burst above its allocated units if no other containers are taking up resources. **For instance**, if you have two tasks running on a `t2.medium`, each with 1024 CPU units budgeted for its respective container, one task could take up the whole 2048 CPU units given that the other task is inactive.
+
+Besides, containers also share their unallocated CPU units with other containers on the instance with the same ratio as their allocated amount. **For example**, if you have two tasks running on a `t2.medium` each with 0 CPU units, each container of each task will effectively get 1024 CPU units since no other container has reserved them.
+
+In addition to such benefits of using ECS containers, the data given in the table above can give input to your choices of EC2 instance types or designing ECS containers on your instances.
+
+Whereas the CPU units can be reserved with soft limit because containers can burst above their provision, the Memory reservation has to stay within the allocated amount. 
+
+**For example**, you need a configuration of 332 memory units. If you choose `t2.micro` as the EC2 instance type which offers 993 memory units, you can fit 2 containers in the instance (993/332 = 2.99 containers). However, if you choose `t2.small` which offers 2001 memory units, the memory resource would be more efficiently used because 2001/332 = 6.02 containers and not using the leftover 0.02 containers would be less wasteful than doing this with the leftover 0.99 containers.
```

Comment:
There is a lot of content, right after the table above, about the CPU metric for the containers, compared to this example. The CPU and how it works is part of the documentation of ECS. But this last example, isn't. I think there should be a more extensive reference or at least more than one examples.

Reply:
**For example**, let's assume that you need a minimum of 332 MBs (memory units) of memory for your container.

Reply:
You don't explain about the proportional usage of CPU and Memory for each instance type. After reading this example I was also left with a sensation "where is the rest of the blog post?". Please split it (virtually at least) into 

- Introduction
- Problem Description
- Main Body (explanation how to solve the problem)
- Conclusion

The above can be virtual, mostly to help you structure the post. You don't need to use section names at all.